### DISCUSSION TITLE
Можно ли реализовать на totum сборщик отчетов?

### DISCUSSION THREAD

**Discussion by wedcanto** [totumonline/totum-ru-issues-and-discussions#397] [Как это сделать?] [answered]
У нас госконтора, в которой много мелких подразделений. И часто начальство придумывает простые таблички, по которым нужно собрать данные, либо собирать их ежедневно/ежемесячно
Подскажите как лучше сделать на totum:

Админ создает таблицу, в которой к примеру вверху три столбца - название продукта, количество в наличии, количество сколько еще необходимо единиц продукта. Нужно чтобы:
1. Каждое подразделение видело и могло редактировать только свои данные
2. Админ видел одну сводную таблицу
3. Админ видел отдельный список подразделений, кто не заполнил отчет за определенный день

Нужно создавать для каждого подразделения свою таблицу? Или можно составить одну, и дать как-то доступ только к строкам соответствующего подразделения? Если для каждого подразделения нужно свою таблицу, очень геморно это...подразделений больше 50 штук.
И как реализовать периодичный сбор этих отчетов, чтобы подразделениям было понятно и очевидно какие таблицы заполнять.

Может сумбурно объяснил, но прошу хотя бы направить в правильном направлении, как лучше организовать.

Авторизация и назначение ролей на основе LDAP (Active Directory)

**Comment by ttm-alexeypolunin** [comment]
Можете пример того, что подразделение заполняет?

**Reply by wedcanto** [reply]
Да ну вот банально, отчет по МТО, по компам

Подразделение; Сколько положено компов; Сколько в наличии; % обеспеченности; Сколько устаревших; Сколько нужно докупить.

Это как пример, но в основном отчеты подобные. Сейчас образец рассылается по почте, собирается в эксельках, и потом один страдалец все обьединяет)

**Reply by ttm-alexeypolunin** [reply, marked as answer]
Я бы представил это так:

Отчет на подразделение — Цикл. Таким образом у вас все отчеты в одной таблице — Циклов.

У циклов есть настройка видимости относительно его владельца – ставите в Видит только владелец.

Что бы разослать отчет — нужно добавить в таблицу Циклов строки с указанием их владельцев.

Отчеты имеют разный состав. В самом простом варианте вы внутрь цикла добавляете в строчную часть нужные строки — одна колонка это параметр, вторая его значение. Какие параметры будут в первой колонке — для этого делаете еще одну таблицу с Циклами, где уже каждый цикл это список параметров. Тут же может быть и отчет по подразделениям и сюда подтягивать значения из того, что они заполнили.

Проблема такого подхода заключается в то, что у Тотума колонки задаются жестко Админом. Те внутри отчета у вас Значение должно быть типом Строка (как я себе представляю). И иногда туда будут писать фигню, которая вам формулы будет ломать. Но можно извернуться и передавать например regexp, запрещающий для числового параметра писать буковки.

Второй подход — если вам нужно более изощренно настраивать отчет, с кодами и прочей фигней — это делать под каждый временную таблицу админом. Далее сделать например Простую таблицу в которую записывается подразделение и отчет — оттуда открывать нужную временную и сохранять обратно все, что они там назаполняли. И потом уже из нее делать отчет.

LDAP — смотрите в доках: https://ru.docs.totum.online/pro-ldap-ad

«и дать как-то доступ только к строкам соответствующего подразделения» — для Простых таблиц это делают Префильтры: https://ru.docs.totum.online/prefilters-basic-level

Короткая консультация бесплатно: https://control.demoru.totum.online/Forms/free-consultation

**Reply by wedcanto** [reply]
Алексей, спасибо большое за подробный ответ. Пока я понял одно - моя задача вполне реализуема на totum, но для этого мне нужно хорошенько изучить всю документацию, тогда смогу понять ваш совет и уже приступить к реализации. 

Ещё раз спасибо, надеюсь что до всего остального смогу дойти самостоятельно.

**Reply by ttm-alexeypolunin** [reply]
Если закопаетесь - пишите в телеграм. Ну и 20 мин консультация - обычно такие все из рекламы состоят, я зачем-то по делу отвечаю :)

**Comment by ttm-alexeypolunin** [comment]
Ок, напишу сег-завтра
